application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/wolframalpha/WolframAlphaCommandUtils.java

```diff
@@ -0,0 +1,226 @@
+package org.togetherjava.tjbot.commands.mathcommands.wolframalpha;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.requests.restaction.WebhookMessageUpdateAction;
+import org.jetbrains.annotations.NotNull;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.font.FontRenderContext;
+import java.awt.geom.AffineTransform;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+enum WolframAlphaCommandUtils {
+    ;
+
+    static byte[] imageToBytes(BufferedImage img) throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ImageIO.write(img, "png", baos);
+        return baos.toByteArray();
+    }
+
+    static BufferedImage combineImages(List<BufferedImage> images, int height) {
+        int width =
+                images.stream().mapToInt(BufferedImage::getWidth).max().orElse(Integer.MAX_VALUE);
+        BufferedImage image = new BufferedImage(width, height, BufferedImage.TYPE_4BYTE_ABGR);
+        Graphics imgGraphics = image.getGraphics();
+        imgGraphics.setColor(Color.WHITE);
+        imgGraphics.fillRect(0, 0, width, height);
+        int resultHeight = 0;
+        for (BufferedImage img : images) {
+            imgGraphics.drawImage(img, 0, resultHeight, Constants.IMAGE_OBSERVER);
+            resultHeight += img.getHeight(Constants.IMAGE_OBSERVER);
+        }
+        return image;
+    }
+
+    static boolean compareImages(BufferedImage img1, BufferedImage img2) {
+        int width1 = img1.getWidth();
+        int height1 = img1.getHeight();
+        return width1 == img2.getWidth() && height1 == img2.getHeight()
+                && IntStream.range(0, width1)
+                    .mapToObj(x -> IntStream.range(0, height1)
+                        .anyMatch(y -> img1.getRGB(x, y) != img2.getRGB(x, y)))
+                    .noneMatch(x -> x);
+    }
+
+    static int getWidth(String header) {
+        return (int) Constants.WOLFRAM_ALPHA_FONT
+            .getStringBounds(header, new FontRenderContext(new AffineTransform(), true, true))
+            .getWidth();
+    }
+
+    static String handleMisunderstoodQuery(QueryResult result) {
+        List<String> output = new ArrayList<>();
```

Comment:
Maybe output is valid, I just don't like it really.
What does it output? If there's something more descriptive you can call it, I'd change it to that

Reply:
If the API does not understand the query this type of result is returned. It has certain tags which contain information for the user about how or why their query was misunerstood